### PR TITLE
add PHP-FPM

### PR DESCRIPTION
**Pull request by VVelox**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

At current time, this is limited to a single pool.

### CODE REVIEW

**Comment by VVelox** [comment]
Poller submitted in...

https://github.com/librenms/librenms-agent/pull/83

File: includes/polling/applications/php-fpm.inc.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?php
+$name = 'php-fpm';
+$app_id = $app['app_id'];
+
+$options      = '-O qv';
+$mib          = 'NET-SNMP-EXTEND-MIB';
+$oid          = '.1.3.6.1.4.1.8072.1.3.2.3.1.2.8.112.104.112.102.112.109.115.112';
+$phpfpm = snmp_walk($device, $oid, $options, $mib);
```
Comment (laf) [inline includes/polling/applications/php-fpm.inc.php]: You don't need $mib if you are using the numerical oid. Can you use a named oid instead?

**Comment by laf** [comment]
Thanks for this, you will also need to update the applications doc to show the use. please pay attention to the standard location we ask people to download the agent scripts to.

File: includes/polling/applications/php-fpm.inc.php [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 $options      = '-O qv';
 $mib          = 'NET-SNMP-EXTEND-MIB';
-$oid          = '.1.3.6.1.4.1.8072.1.3.2.3.1.2.8.112.104.112.102.112.109.115.112';
+$oid          = 'NET-SNMP-EXTEND-MIB::nsExtendOutputFull.8.112.104.112.102.112.109.115.112';
```
Comment (laf) [inline includes/polling/applications/php-fpm.inc.php]: Thanks but you don't or shouldn't need NET-SNMP-EXTEND-MIB:: as you've already specified the mib above.

File: includes/polling/applications/php-fpm.inc.php (hunk 1 above)
Comment (VVelox) [inline includes/polling/applications/php-fpm.inc.php]: Sweet. Updated it to use a NET-SNMP-EXTEND-MIB::nsExtendOutputFull.8.112.104.112.102.112.109.115.112 .

**Comment by VVelox** [comment]
Okay. Docs updated. :)

File: includes/polling/applications/php-fpm.inc.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<?php
+$name = 'php-fpm';
+$app_id = $app['app_id'];
+
+$options      = '-O qv';
+$mib          = 'NET-SNMP-EXTEND-MIB';
+$oid          = 'NET-SNMP-EXTEND-MIB::nsExtendOutputFull.8.112.104.112.102.112.109.115.112';
```
Comment (laf) [inline includes/polling/applications/php-fpm.inc.php]: Just need to remove `NET-SNMP-EXTEND-MIB::`

File: doc/Extensions/Applications.md [hunk 4]
```diff
@@ -469,3 +470,20 @@ if [ "$1" = "config" ]; then
 fi
 echo -n "foobar.value " $(date +%s) #Populate a value, here unix-timestamp
 ```
+
+
+#### PHP-FPM
+
+##### SNMP Extend
+
+1. Copy the shell script, phpfpm-sp, to the desired host (the host must be added to LibreNMS devices)
```
Comment (laf) [inline doc/Extensions/Applications.md]: Please add the command to download the file like: https://github.com/VVelox/librenms/blob/3a10929a8f62652cc1fc846f38c1da98395bbfd2/doc/Extensions/Applications.md#proxmox

**Comment by florianbeer** [comment]
Seems like this doesn't update `/apps/app=php-fpm/`.
Anyone else seeing this issue or ist something wrong on my end?

**Comment by VVelox** [comment]
Does walking nsExtendOutputFull.8.112.104.112.102.112.109.115.112 result in something like....

NET-SNMP-EXTEND-MIB::nsExtendOutputFull."phpfpmsp" = STRING: www
16/Feb/2017:02:45:33 -0600
112243
127805
0
0
0
2
1
3
20
11
0

Then you hae the snmpd stuff either setup wrong. As long as that is returning that, it should work.

**Comment by florianbeer** [comment]
Sure, everything else works like it should. There are just no graphs on the application overview page (`$URL/apps/app=php-fpm/`).

Btw. thanks for implementing this, I like it very much :+1:

**Comment by VVelox** [comment]
Ahh! Okay. I see what your saying now.

Hmm... Odd. I honestly have no clue. I implemented it based off of the DHCP Stats one which works there. I can't find any notable differences between the two outside of a change of name and some RRD stuff. :(

**Comment by florianbeer** [comment]
Does it work on your installation? I've only got header panels per server (that has the application activated) but no graphs under it.

Maybe the output of `$URL/apps/app=php-fpm/?debug=true` (click on "Show PHP Debug" in the footer) can give you any clues?

**Comment by VVelox** [comment]
Yeah, it works if I go to a device in question and the go to the apps tab and it.

The debug stuff gives me the items below. Nothing really of use off hand though.

Array ( [errno] => 2 [errstr] => file_get_contents(/proc/sys/kernel/hostname): failed to open stream: No such file or directory [errfile] => /usr/local/www/librenms/includes/defaults.inc.php [errline] => 797 )
Array ( [errno] => 8 [errstr] => Undefined index: delete [errfile] => /usr/local/www/librenms/includes/functions.php [errline] => 1829 )
Array ( [errno] => 8 [errstr] => Undefined index: delete_sec [errfile] => /usr/local/www/librenms/includes/functions.php [errline] => 1830 )
Array ( [errno] => 8 [errstr] => Undefined index: sess_id [errfile] => /usr/local/www/librenms/html/includes/authenticate.inc.php [errline] => 56 )
Array ( [errno] => 8 [errstr] => Undefined index: token [errfile] => /usr/local/www/librenms/html/includes/authenticate.inc.php [errline] => 56 )
Array ( [errno] => 8 [errstr] => Undefined offset: 1 [errfile] => /usr/local/www/librenms/html/includes/authentication/mysql.inc.php [errline] => 40 )
Array ( [errno] => 8 [errstr] => Undefined index: custom_css [errfile] => /usr/local/www/librenms/html/index.php [errline] => 147 )
Array ( [errno] => 8 [errstr] => Undefined index: menu [errfile] => /usr/local/www/librenms/LibreNMS/Plugins.php [errline] => 82 )
Array ( [errno] => 8 [errstr] => Undefined variable: device [errfile] => /usr/local/www/librenms/html/pages/apps.inc.php [errline] => 137 )
Array ( [errno] => 8 [errstr] => Undefined variable: sep [errfile] => /usr/local/www/librenms/html/pages/apps.inc.php [errline] => 142 )
Array ( [errno] => 8 [errstr] => Undefined index: php-fpm [errfile] => /usr/local/www/librenms/html/pages/apps/default.inc.php [errline] => 25 )
Array ( [errno] => 2 [errstr] => Invalid argument supplied for foreach() [errfile] => /usr/local/www/librenms/html/pages/apps/default.inc.php [errline] => 25 )
Array ( [errno] => 8 [errstr] => Undefined index: php-fpm [errfile] => /usr/local/www/librenms/html/pages/apps/default.inc.php [errline] => 25 )
Array ( [errno] => 2 [errstr] => Invalid argument supplied for foreach() [errfile] => /usr/local/www/librenms/html/pages/apps/default.inc.php [errline] => 25 )

**Comment by VVelox** [comment]
Well those errors appear to be harmless. I am seeing them on most pages I bring up.

**Comment by florianbeer** [comment]
Got it and PR'd it #5937

**Pull request merged**